parsers/handleWordpressComponents.js

```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
   };
 }
 
+const findCaptionText = (figureCaption, key) => {
+  let figCaptionValue = '';
+  const nestedFigCap = figureCaption.filter((o) => o.type === key);
+  if (nestedFigCap.length > 0) {
+    figCaptionValue = nestedFigCap[0].value;
+  } else {
+    const string = findCaptionText(figureCaption[0].children, key);
+    if (string) {
+      figCaptionValue += string;
+    }
+  }
+  return figCaptionValue;
+};
+
+export function findImageData() {
+  return (tree) => {
+    visit(tree, (node) => node.type === 'element' && node.children.some((n) => n.tagName === 'img'), (node) => {
+      const arrayOfImageData = node.children.filter((imageData) => imageData.tagName === 'img');
+      const { src } = arrayOfImageData[0].properties;
+      const { alt } = arrayOfImageData[0].properties;
+
+      const figureCaption = node.children.filter((element) => element.tagName === 'figcaption');
+
+      let captionText = '';
+      if (figureCaption.length > 0) {
+        captionText = findCaptionText(figureCaption, 'text');
+      } else captionText = '';
```

Comment:
That line seems excessive - `captionText` is already defined as empty string three lines up.

Reply:
Definitely excessive!